entity-framework/core/miscellaneous/multitenancy.md

```diff
@@ -0,0 +1,148 @@
+
+- **Tenant1** - `tenant1.CustomerData`
+- **Tenant2** - `tenant2.CustomerData`
+
+This can be supported with some extra effort. For example, you could use `OnModelCreating` to specify the schema.
+
+```csharp
+private readonly string tenant = string.Empty;
+
+public MultiSchemaContext(
+    DbContextOptions<SingleDbContext> options,
+    TenantProvider tenantProvider)
+    : base(options) 
+{
+    tenant = tenantProvider.GetTenant();
+}
+
+public DbSet<CustomerData> Data { get; set; }
+
+protected override void OnModelCreating(ModelBuilder modelBuilder)
+{
+    modelBuilder.Entity<CustomerData>().ToTable(nameof(CustomerData), tenant);
+}
+```
+
+Another approach is to define the table as a base class, then derive a class for each tenant. A `DbSet` is defined for every tenant and a dictionary can be used to provide generic access to the base class by tenant.
+
+## Multiple databases and connection strings
+
+The multiple database version is implemented by passing a different connection string for each tenant. This can be configured at startup by resolving the service provider and using it to build the connection string. This example specifies a different file by tenant for SQlite, but can easily be repurposed to specify a different database name in a SQL Server or other provider's connection string:
+
+```csharp
+services.AddDbContextFactory<MultipleDbContext>((sp, opts) =>
+{
+    var tenantProvider = sp.GetRequiredService<TenantProvider>();
+    opts.UseSqlite($"Data Source={tenantProvider.GetTenant()}.sqlite");
+}, ServiceLifetime.Scoped);
+```
+
+This works fine for most scenarios unless the user can switch tenants in realtime.
+
+### Realtime tenant changes
+
+In the previous configuration for multiple databases, the options are cached at the `Scoped` level. This means that if the user changes the tenant, the options are _not_ reevaluated and so the tenant change isn't reflected in queries.
+
+The easy solution for this when the tenant _can_ change is to set the lifetime to `Transient.` This ensures the tenant is re-evaluated along with the connection string each time a `DbContext` is requested. The user can switch tenants as often as they like. The following table helps you choose which lifetime makes the most sense for your factory.
+
+|**Scenario**|**Single database**|**Multiple databases**|
+|:--|:--|:--|
+|_User stays in a single tenant_|`Scoped`|`Scoped`|
+|_User can switch tenants_|`Scoped`|`Transient`|
+
+The default of `Singleton` still makes sense if your database does not take on user-scoped dependencies.
+
+## Performance notes
+
+It is highly unlikely that a single user session will require hundreds or thousands of `DbContext` instances during a given session. In local benchmarks it only takes 3 microseconds on a laptop to go from requesting the factory to creating a usable `DbContext`. That translates to over 300,000 requests per second.
```

Comment:
If it really does take only 3 microseconds, I'll bet that's the case only with DbContext pooling - or did you see something else?

In any case, it may be good to clarify what exactly you're advising users, and how that's specific to multitenancy.